### PR TITLE
Migrate codebase to Kotlin

### PR DESCRIPTION
**Pull request by adamszadkowski**

### CODE REVIEW

File: src/test/kotlin/info/szadkowski/contact/service/mail/MailMessageServiceTest.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Mock
+    lateinit var mimeMessage: MimeMessage
+
+    @BeforeEach
+    fun setUp() {
+        val mailAddressesProperties = MailAddressesProperties()
+        mailAddressesProperties.senderMail = "sender@address.com"
+        mailAddressesProperties.recipientMail = "recipient@address.com"
+        Mockito.`when`(sender.createMimeMessage()).thenReturn(mimeMessage)
```
Comment (sobolewskikamil) [inline src/test/kotlin/info/szadkowski/contact/service/mail/MailMessageServiceTest.kt]: You can check out mockk library. Plays really well with Kotlin.

File: build.gradle.kts [hunk 2]
```diff
@@ -34,21 +35,28 @@ java {
   targetCompatibility = JavaVersion.VERSION_11
 }
 
+val compileKotlin: KotlinCompile by tasks
```
Comment (sobolewskikamil) [inline build.gradle.kts]: You can also write something like:
`tasks.withType<KotlinCompile>().configureEach {
    kotlinOptions.jvmTarget = "1.8"
}`

File: src/intTest/kotlin/info/szadkowski/contact/ContactServiceApplicationTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    lateinit var throttlingProperties: ThrottlingProperties
+
+    @Autowired
+    lateinit var templateProperties: TemplateProperties
+
+    @Test
+    fun `Should inject main beans`() {
+        assertThat(controller).isNotNull()
```
[the remark this answers is not in the document]
Comment (adamszadkowski) [reply src/intTest/kotlin/info/szadkowski/contact/ContactServiceApplicationTest.kt]: I rather like version with explicit method execution. It is a little more obvious that assertion is done there

File: src/main/kotlin/info/szadkowski/contact/configuration/ThrottlerConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    lateinit var properties: ThrottlingProperties
+
+    @Bean
+    fun timeProvider() = SystemTimeProvider()
+
+    @Bean
+    fun throttlerFactory(timeProvider: TimeProvider) = ThrottlerFactory(timeProvider)
+
+    @Bean
+    fun ipThrottler(throttlerFactory: ThrottlerFactory) = createThrottler(throttlerFactory, properties.ip!!)
```
[the remark this answers is not in the document]
Comment (adamszadkowski) [reply src/main/kotlin/info/szadkowski/contact/configuration/ThrottlerConfiguration.kt]: It is hard to do this - someone would have to put his own `application.yml` into container. Many other things probably would broke. Throwing some kind of exception is currently good enough.

File: build.gradle.kts (hunk 2 above)
Comment (adamszadkowski) [reply build.gradle.kts]: It in fact looks better.

File: src/test/kotlin/info/szadkowski/contact/service/mail/MailMessageServiceTest.kt (hunk 1 above)
Comment (adamszadkowski) [reply src/test/kotlin/info/szadkowski/contact/service/mail/MailMessageServiceTest.kt]: I like it :)

**Pull request merged**